test/core/origin_core/propose_block.js

```diff
@@ -0,0 +1,302 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+// Test: propose_block.js
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const web3 = require('../../test_lib/web3.js');
+
+const BN = require('bn.js');
+const EventsDecoder = require('../../test_lib/event_decoder.js');
+const Utils = require('../../test_lib/utils.js');
+
+const OriginCore = artifacts.require('OriginCore');
+
+contract('propose meta-block', async (accounts) => {
```

Comment:
Having a naming convention like this makes it easier to find the code that corresponds to a test-case. Otherwise the test won't output the contract name.
```suggestion
contract('OriginCore.proposeBlock()', async (accounts) => {
```